src/Components/Search/Spike/SearchBarSpike.tsx

```diff
@@ -0,0 +1,226 @@
+import { ContextConsumer, ContextProps } from "Artsy"
+import Input from "Components/Input"
+import { SearchPreview } from "Components/Search/Previews"
+import { throttle } from "lodash"
+import React, { Component } from "react"
+import Autosuggest from "react-autosuggest"
+import {
+  createRefetchContainer,
+  graphql,
+  QueryRenderer,
+  RelayRefetchProp,
+} from "react-relay"
+import { get } from "Utils/get"
+
+export interface Props extends ContextProps {
+  relay: RelayRefetchProp
+  viewer: SearchBarSpike_viewer
+}
+
+interface State {
+  /* Holds current input */
+  input: string
+  /* For preview generation of selected items */
+  entityID: string
+  entityType: string
+}
+
+export class SearchBarSpike extends Component<Props, State> {
+  state = {
+    input: "",
+    entityID: null,
+    entityType: null,
+  }
+
+  // Throttled method to toggle previews.
+  throttledOnSuggestionHighlighted = ({ suggestion }) => {
+    if (!suggestion) return null
+    const {
+      node: { searchableType: entityType, id: entityID },
+    } = suggestion
+    this.setState({ entityType, entityID })
+  }
+
+  // Throttled method to perform refetch for new suggest query.
+  throttledFetch = ({ value: term }) => {
+    this.props.relay.refetch(
+      {
+        term,
+        hasTerm: true,
+      },
+      null,
+      error => {
+        if (error) console.error(error)
+      }
+    )
+  }
+
+  componentDidMount() {
+    this.throttledFetch = throttle(this.throttledFetch, 500, {
+      leading: true,
+    })
+    this.throttledOnSuggestionHighlighted = throttle(
+      this.throttledOnSuggestionHighlighted,
+      500,
+      { leading: true }
+    )
+  }
+
+  searchTextChanged = (_e, { newValue: input }) => {
+    this.setState({ input })
+  }
+
+  onSuggestionsClearRequested = () => {
+    this.setState({ input: "", entityID: null, entityType: null })
+  }
+
+  // Navigate to selected search item.
+  onSuggestionSelected = (
+    _e,
+    {
+      suggestion: {
+        node: { href },
+      },
+    }
+  ) => {
+    window.location.href = href
+  }
+
+  renderPreview() {
+    const { entityID, entityType } = this.state
+    if (entityID && entityType) {
+      return <SearchPreview entityID={entityID} entityType={entityType} />
+    }
+  }
+
+  renderSuggestionsContainer = ({ containerProps, children, query }) => {
+    if (!query) return null
+    return (
+      <Box {...containerProps}>
+        <Flex flexDirection={["column", "row"]}>
+          <Box width={["100%", "25%"]}>
+            <Flex flexDirection="column">
+              <Box>Search "{query}"</Box>
+              {children}
+            </Flex>
+          </Box>
+          <Box width={["100%", "75%"]}>{this.renderPreview()}</Box>
+        </Flex>
+      </Box>
+    )
+  }
+
+  renderSuggestion = ({ node: { displayLabel } }, { isHighlighted }) => {
+    return (
+      <span style={{ backgroundColor: isHighlighted ? "#ddd" : "#fff" }}>
```

Comment:
This is just temporary, subsequent PR will make this a proper little `SuggestionItem`.